tool/update_bindings.dart

```diff
@@ -175,28 +176,35 @@ Future<void> _generateJsTypeSupertypes() async {
   final dartJsInterop = await contextCollection.contexts.single.currentSession
       .getLibraryByUri('dart:js_interop') as LibraryElementResult;
   final definedNames = dartJsInterop.element.exportNamespace.definedNames;
-  final jsTypeSupertypes = <String, String?>{};
+  // `SplayTreeMap` to avoid moving types around in `dart:js_interop` affecting
+  // the code generation.
+  final jsTypeSupertypes = SplayTreeMap<String, String?>();
   for (final name in definedNames.keys) {
     final element = definedNames[name];
     if (element is TypeDefiningElement) {
+      // TODO(srujzs): This contains code that handles the SDK before and after
+      // the migration of JS types to extension types. Once the changes to
+      // migrate to extension types hit the dev branch, we should remove some of
+      // the old code.
       void storeSupertypes(InterfaceElement element) {
-        bool isInJsTypes(InterfaceElement element) =>
+        bool isInJsTypesOrJsInterop(InterfaceElement element) =>
```

Comment:
double-nested function? Necessary? 

Could be moved to top-level?

Reply:
Not necessary, done! I like local functions because they're declared right next to where they're used, but I don't know if that's against dart style.